jerry-core/ecma/builtin-objects/ecma-builtin-global.cpp

```diff
@@ -1164,6 +1164,149 @@ ecma_builtin_global_object_encode_uri (ecma_value_t this_arg __attr_unused___, /
   return ecma_builtin_global_object_encode_uri_helper (uri, unescaped_uri_set);
 } /* ecma_builtin_global_object_encode_uri */
 
+/*
+ * Maximum value of a byte.
+ */
+#define ESCAPE_MAXIMUM_BYTE_VALUE (255)
+
+/*
+ * Format is a percent sign followed by lowercase u and four hex digits.
+ */
+#define ESCAPE_ENCODED_UNICODE_CHARACTER_SIZE (6)
+
+/*
+ * Escape characters bitset:
+ *   One bit for each character between 0 - 127.
+ *   Bit is set if the character does not need to be converted to %xx form.
+ *   These characters are: a-z A-Z 0-9 @ * _ + - . /
+ */
+static const uint8_t escape_set[16] =
+{
+  0x0, 0x0, 0x0, 0x0, 0x0, 0xec, 0xff, 0x3,
```

Comment:
Where did you get this constants? RFC 3986?

How does this list corresponds to the following from spec:
```
uriAlpha ::: one of
a b c d e f g h i j k l m n o p q r s t u v w x y z
A B C D E F G H I J K L M N O P Q R S T U V W X Y Z
uriMark ::: one of
- _ . ! ~ * ' ( )
```